fcrepo4/fcrepo4
# vcard:member predicates with URI objects are now recognized as author…

…ized users

Resolves:

**Now URI objects of vcard:member predicates in lists referenced by acl:agentGroup predicates are properly authroized.**
* * *

**JIRA Ticket**:  https://jira.duraspace.org/browse/FCREPO-2921

# What does this Pull Request do?
If you have configured your Fedora instance to use a "fcrepo.auth.webac.userAgent.baseUri" system property,  you can now expect that  a <> vcard:member <${fcrepo.auth.webac.userAgent.baseUri}/your-username> triple in your group list referenced by acl:agentGroup in your acl will be properly authorized.

I have added an integration test to confirm that this works.

# How should this be tested?
1. set the system property of your fedora instance: 
`-Dfcrepo.auth.webac.userAgent.baseUri=http://example.com/`
2. create a resource 
`curl -u fedoraAdmin:fedoraAdmin -X PUT http://localhost:8080/rest/test`
3. create a group-list resource with the following body: 
group-list.ttl
```
@prefix    acl:  <http://www.w3.org/ns/auth/acl#>.
@prefix  vcard:  <http://www.w3.org/2006/vcard/ns#>.
<> a vcard:Group ;
          vcard:hasMember  <http://example.com/testuser> .
```
`curl -u fedoraAdmin:fedoraAdmin -X PUT http://localhost:8080/rest/group-list -H "Content-Type: text/turtle" --data "@group-list.ttl"`

4. create an acl for the test resource with the following body (acl.ttl)
acl.ttl
```
@prefix acl: <http://www.w3.org/ns/auth/acl#> .

<#authorization>    a               acl:Authorization;
    acl:accessTo    </rest/test>;
    acl:mode        acl:Read;
    acl:mode        acl:Write;
    acl:agentGroup  </rest/group-list>.
```

`curl -u fedoraAdmin:fedoraAdmin -X PUT http://localhost:8080/rest/test/fcr:acl -H "Content-Type: text/turtle" --data "@acl.ttl"`

5. Verify the resource is read accessible with the testuser.
`curl -i -u testuser:testpass http://localhost:8080/rest/test`
6. Verify the resource is write accessible with the testuser.
`curl -i -u testuser:testpass -X PUT http://localhost:8080/rest/test/child`

# Interested parties
Tag (@ mention) interested parties or, if unsure, @fcrepo4/committers
